homeassistant/components/profiler/__init__.py

```diff
@@ -59,22 +60,27 @@ async def _async_generate_profile(hass: HomeAssistant, call: ServiceCall):
     )
     profiler = cProfile.Profile()
     profiler.enable()
+    heap_profiler = hpy()
+    heap_profiler.setref()
     await asyncio.sleep(float(call.data[CONF_SECONDS]))
     profiler.disable()
+    heap = heap_profiler.heap()
```

Comment:
The heap profiler is likely going to make the cProfile more difficult to follow since it will have an impact that cProfile can't exclude.

I think it should have its own service.

Reply:
Fair enough!

Reply:
maybe `profiler.start_guppy3`

Reply:
I was thinking something simpler like `profiler.memory`

Reply:
Also, we can do a couple things - Capture the 60 second heap diff like implemented here, or just capture the whole heap. I was hoping exporting to the file would give you the ability to drill down into the tree, but it doesn't appear to do so like you can with a `pdb` session looking at the sets on the object...